extensions/cms/src/cmsUtils.ts

```diff
@@ -143,63 +185,112 @@ export class CmsUtils {
-		return provider.removeRegisteredServer(ownerUri, relativePath, registeredServerName).then((result) => {
-			return result;
-		});
+		let result = await provider.removeRegisteredServer(ownerUri, relativePath, registeredServerName);
+		return result;
 	}
 
 	public async addServerGroup(groupName: string, groupDescription: string, relativePath: string, ownerUri: string): Promise<boolean> {
 		let provider = await this.getCmsService();
-		return provider.addServerGroup(ownerUri, relativePath, groupName, groupDescription).then((result) => {
-			return result;
-		});
+		let result = await provider.addServerGroup(ownerUri, relativePath, groupName, groupDescription);
+		return result;
 	}
 
 	public async removeServerGroup(groupName: string, relativePath: string, ownerUri: string): Promise<boolean> {
 		let provider = await this.getCmsService();
-		return provider.removeServerGroup(ownerUri, relativePath, groupName).then((result) => {
-			return result;
-		});
+		let result = await provider.removeServerGroup(ownerUri, relativePath, groupName);
+		return result;
 	}
 
 	// Getters
 	public get registeredCmsServers(): ICmsResourceNodeInfo[] {
 		return this._registeredCmsServers;
 	}
 
-	public get connection(): Thenable<azdata.connection.Connection> {
-		return this.openConnectionDialog([cmsProvider], undefined, { saveConnection: false }).then((connection) => {
-			if (connection) {
-				// remove group ID from connection if a user chose connection
-				// from the recent connections list
-				connection.options['groupId'] = null;
-				connection.providerName = mssqlProvider;
-				return connection;
+	public async credentialProvider(): Promise<azdata.CredentialProvider> {
+		if (!this._credentialProvider) {
+			this._credentialProvider = await azdata.credentials.getProvider(CredentialNamespace);
+		}
+		return this._credentialProvider;
+	}
+
+	public async connection(initialConnectionProfile?: azdata.IConnectionProfile): Promise<azdata.connection.Connection> {
+		if (!initialConnectionProfile) {
+			initialConnectionProfile = {
+				connectionName: undefined,
+				serverName: undefined,
+				databaseName: undefined,
+				userName: undefined,
+				password: undefined,
+				authenticationType: undefined,
+				savePassword: undefined,
+				groupFullName: undefined,
+				groupId: undefined,
+				providerName: undefined,
+				saveProfile: undefined,
+				id: undefined,
+				options: {}
+			};
+		}
+		let connection = await this.openConnectionDialog([cmsProvider], initialConnectionProfile, { saveConnection: false });
+		if (connection) {
+			// remove group ID from connection if a user chose connection
+			// from the recent connections list
+			connection.options['groupId'] = null;
+			connection.providerName = mssqlProvider;
+			if (connection.options.savePassword) {
+				await this.savePassword(connection.options.user, connection.options.password);
 			}
-		});
+			return connection;
+		} else {
+			return Promise.reject();
+		}
```

Comment:
You're still not providing a message here. If this is truly an error (which rejecting implies it is) then you should provide some sort of message so callers up the stack can have some sort of information if the error gets to them.